### ISSUE TITLE
[DEPR]: Matlab API Key

### ISSUE THREAD

**Issue by ormsbee** [openedx/public-engineering#192]
### Overview

Our system currently has special handling for MATLAB problems via an old integration through xqueue that isn't really supported any longer. Since this integration should now be done via LTI, we can remove:

* [`matlab_api_key`](https://github.com/openedx/edx-platform/blob/master/xmodule/capa_block.py#L333-L341) attribute in ProblemBlock, as well as the version [defined for inheritance purposes](https://github.com/openedx/edx-platform/blob/f8737d343980ca8531d65c41be3b982e8333dc93/xmodule/modulestore/inheritance.py#L148-L155).
* [`MatlabInput`](https://github.com/openedx/edx-platform/blob/master/xmodule/capa/inputtypes.py#L839-L1032)
  * Associated JavaScript, CSS, template, and text fixture code.
  * It may be that removing this would adversely affect grades, in which case it may be better to leave behind a bare shell that does nothing but acting as a dumb placeholder for legacy content scoring purposes. More investigation needed.

Affect on old course content:

* Student scores must be preserved.
* It is not necessary to be able to render the original student submission or detailed response information.

### Proposal Date

10 May 2023

### Target Ticket Acceptance Date

24 May 2023

### Earliest Open edX Named Release Without This Functionality

**Quince**.

Since it is not currently in a working state, there's no need to offer an extra release for migration.

### Rationale

1. The current endpoint does not hook up to a working external service, and is not useful.
2. LTI is already the accepted way to integrate MATLAB content.
3. It will slightly simplify ProblemBlock and the LMS's XBlock runtime to remove this.

**Comment by ormsbee** [comment]
There was a [slack thread](https://openedx.slack.com/archives/CR0UMU0QP/p1683746232934579) about this as well.

**Comment by kdmccormick** [comment]
Just to be clear @ormsbee , would this DEPR also cover the Matlab API Key field in the problem editor, thus clearing up 2U's T&L team to remove it from the new React-based problem editor? (cc @jristau1984)

**Comment by ormsbee** [comment]
@kdmccormick: Yes, but to be clear: I don't think acceptance this DEPR should gate that work, since it's more of a bookkeeping DEPR–i.e. I'm pretty sure that it's been broken for years, so immediately removing the view of the broken thing doesn't negatively impact users. I'm leaving it open for a couple of weeks of comment Just In Case, and if people want to add any more information.

**Comment by dianakhuang** [comment]
@ormsbee is this DEPR complete now?

**Comment by ormsbee** [comment]
@dianakhuang: No, its not. I removed it from the settings interface so that people wouldn't see it anymore, but I didn't remove the underlying code because a lot of capa's xqueue interface code uses MATLAB for its tests. So if I removed MATLAB, I'd have to re-implement those tests for xqueue, which didn't seem worthwhile since xqueue itself was on the chopping block. I'm now thinking that this DEPR will be executed when the xqueue code in edx-platform finally dies as part of https://github.com/openedx/public-engineering/issues/286

**Comment by dianakhuang** [comment]
Blocked on the XQueue ticket: https://github.com/openedx/public-engineering/issues/286

**Comment by feanil** [comment]
Given that the old implementation doesn't work, this can be considered unblocked for removal.  We don't need to work on it separately if the removal of all the xqueue related code is close to landing but this can be picked up for removal sooner if that's gonna be a while.

**Comment by ormsbee** [comment]
@feanil: The FC work for xqueue removal was restarted this morning, so I'd ask that this still wait behind that work.

**Comment by feanil** [comment]
Ok, I'll move this back to `Plan Accepted` for now then since the XQueue work is still in progress.
